File: TILAB/src/main/java/com/TILab/TILAB/model/KitComponente.java
package com.TILab.TILAB.model;

import jakarta.persistence.*;

@Entity
@Table(name = "kit_componentes")
public class KitComponente {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "kit_id", nullable = false)
    private Kit kit;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "componente_id", nullable = false)
    private Componente componente;
    
    @Column(nullable = false)
    private Integer cantidad;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Kit getKit() {
        return kit;
    }

    public void setKit(Kit kit) {
        this.kit = kit;
    }

    public Componente getComponente() {
        return componente;
    }

    public void setComponente(Componente componente) {
        this.componente = componente;
    }

    public Integer getCantidad() {
        return cantidad;
    }

    public void setCantidad(Integer cantidad) {
        this.cantidad = cantidad;
    }
}

File: ti-lab-back/app/models/kit_componente.py
from sqlalchemy import Column, Integer, ForeignKey
from sqlalchemy.orm import relationship
from app.models.base import Base


class KitComponente(Base):
    __tablename__ = "kit_componentes"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    kit_id = Column(Integer, ForeignKey("kits.id"), nullable=False)
    componente_id = Column(Integer, ForeignKey("componentes.id"), nullable=False)
    cantidad = Column(Integer, nullable=False)
    
    kit = relationship("Kit", back_populates="kit_componentes")
    componente = relationship("Componente", back_populates="kit_componentes")